crates/engine-primitives/src/traits.rs

```diff
@@ -154,3 +157,21 @@ impl PayloadAttributes for OptimismPayloadAttributes {
         Ok(())
     }
 }
+
+/// The execution payload envelope type.
+pub trait PayloadEnvelopeV3: Send + Sync + std::fmt::Debug {
+    /// Returns execution payload from envelope
+    fn execution_payload(&self) -> ExecutionPayloadV3;
+}
+
+impl PayloadEnvelopeV3 for OptimismExecutionPayloadEnvelopeV3 {
+    fn execution_payload(&self) -> ExecutionPayloadV3 {
+        self.execution_payload.clone()
+    }
+}
+
+impl PayloadEnvelopeV3 for ExecutionPayloadEnvelopeV3 {
+    fn execution_payload(&self) -> ExecutionPayloadV3 {
+        self.execution_payload.clone()
+    }
+}
```

Comment:
this a restriction and contra to why abstracted the engine types in the first place

we can move this to e2e crate instead as a helper trait